tests/run-make/core-ffi-typecheck-clang/rmake.rs

```diff
@@ -0,0 +1,164 @@
+//@ needs-force-clang-based-tests
+
+use run_make_support::{clang, regex, rfs, rustc};
+
+const SKIPPED_TARGETS: &[&str] = &[
+    "riscv",  //error: unknown target triple 'riscv32e-unknown-none-elf'
+    "wasm",   //error: unknown target triple 'wasm32v1-none'
+    "xtensa", //error: unknown target triple 'xtensa-esp32-espidf'
+];
+
+fn main() {
+    let targets = get_target_list();
+
+    let minicore_path = run_make_support::source_root().join("tests/auxiliary/minicore.rs");
+
+    regex_mod();
+
+    for target in targets.lines() {
+        if SKIPPED_TARGETS.iter().any(|prefix| target.starts_with(prefix)) {
+            continue;
+        }
+
+        let clang_output =
+            clang().args(&["-E", "-dM", "-x", "c", "/dev/null", "-target", target]).run();
+
+        let defines = String::from_utf8(clang_output.stdout()).expect("Invalid UTF-8");
+
+        let minicore_content = rfs::read_to_string(&minicore_path);
+        let mut rmake_content = format!(
+            r#"
+            #![no_std]
+            #![no_core]
+            #![feature(link_cfg)]
+            #![allow(unused)]
+            #![crate_type = "rlib"]
+            {}
+            #[path = "processed_mod.rs"]
+            mod ffi;
+            #[path = "tests.rs"]
+            mod tests;
+            "#,
+            minicore_content
+        );
+
+        rmake_content.push_str(&format!(
+            "
+            const CLANG_C_CHAR_SIZE: usize = {};
+            const CLANG_C_CHAR_SIGNED: bool = {};
+            const CLANG_C_SHORT_SIZE: usize = {};
+            const CLANG_C_INT_SIZE: usize = {};
+            const CLANG_C_LONG_SIZE: usize = {};
+            const CLANG_C_LONGLONG_SIZE: usize = {};
+            const CLANG_C_FLOAT_SIZE: usize = {};
+            const CLANG_C_DOUBLE_SIZE: usize = {};
+            ",
+            parse_size(&defines, "CHAR"),
+            parse_signed(&defines, "CHAR"),
+            parse_size(&defines, "SHORT"),
+            parse_size(&defines, "INT"),
+            parse_size(&defines, "LONG"),
+            parse_size(&defines, "LONG_LONG"),
+            parse_size(&defines, "FLOAT"),
+            parse_size(&defines, "DOUBLE"),
+        ));
+
+        // Write to target-specific rmake file
+        let mut file_name = format!("{}_rmake.rs", target.replace("-", "_"));
+
+        if target.starts_with("thumbv8m") {
+            file_name = String::from("thumbv8m_rmake.rs");
+        }
+
+        rfs::create_file(&file_name);
+        rfs::write(&file_name, rmake_content);
+        let rustc_output = rustc()
+            .arg("-Zunstable-options")
+            .arg("--emit=metadata")
+            .arg("--target")
+            .arg(target)
+            .arg(&file_name)
+            .run();
```

Comment:
Passing `-o-` should make it emit to stdout rather than writing a file. Also `write` should create the file if it does not exist.

```suggestion
        // Attempt to build the test file for the relevant target. Tests use constant evaluation,
        // so running is not necessary.
        rfs::write(&file_name, rmake_content);
        let rustc_output = rustc()
            .arg("-Zunstable-options")
            .arg("--emit=metadata")
            .arg("--target")
            .arg(target)
            .arg("-o-")
            .arg(&file_name)
            .run();
```